Src/DependencyCollector/Shared.Tests/Implementation/EventHubsDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,168 @@
+                Assert.AreEqual("ehname", telemetry.Properties["eh.event_hub_name"]);
+                Assert.AreEqual("SomePartitionKeyHere", telemetry.Properties["eh.partition_key"]);
+                Assert.AreEqual("EventHubClient1(ehname)", telemetry.Properties["eh.client_id"]);
+            }
+        }
+
+        [TestMethod]
+        public void EventHubsSendExceptionsAreIgnored()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                this.configuration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+                module.IncludeDiagnosticSourceActivities.Add("Microsoft.Azure.EventHubs");
+                module.Initialize(this.configuration);
+
+                DiagnosticListener listener = new DiagnosticListener("Microsoft.Azure.EventHubs");
+
+                Activity parentActivity = new Activity("parent").AddBaggage("k1", "v1").Start();
+                if (listener.IsEnabled("Microsoft.Azure.EventHubs.Send.Exception"))
+                {
+                    listener.Write("Microsoft.Azure.EventHubs.Send.Exception", new { Exception = new Exception("123") });
+                }
+
+                Assert.IsFalse(this.sentItems.Any());
+            }
+        }
+
+        private T TrackOperation<T>(DiagnosticListener listener, string activityName, TaskStatus status) where T : OperationTelemetry
+        {
+            Activity activity = null;
+
+            if (listener.IsEnabled(activityName))
+            {
+                activity = new Activity(activityName);
+                activity.AddTag("peer.hostname", "eventhubname.servicebus.windows.net");
+                activity.AddTag("eh.event_hub_name", "ehname");
+                activity.AddTag("eh.partition_key", "SomePartitionKeyHere");
+                activity.AddTag("eh.client_id", "EventHubClient1(ehname)");
+                if (listener.IsEnabled(activityName + ".Start"))
+                {
+                    listener.StartActivity(
+                        activity,
+                        new
+                        {
+                            Entity = "ehname",
+                            Endpoint = new Uri("sb://eventhubname.servicebus.windows.net/"),
+                            PartitionKey = "SomePartitionKeyHere"
+                        });
+                }
+                else
+                {
+                    activity.Start();
+                }
+            }
+
+            if (activity != null)
+            {
+                listener.StopActivity(
+                    activity,
+                    new
+                    {
+                        Entity = "ehname",
+                        Endpoint = new Uri("sb://eventhubname.servicebus.windows.net/"),
+                        PartitionKey = "SomePartitionKeyHere",
+                        Status = status
+                    });
+                return this.sentItems.Last() as T;
```

Comment:
Will this miss a case where duplicate telemetry was produced because it only takes the last?
Would it be better to clear `sentItems` between tests and look for `Single()`?
I'm asking because I remember this biting us in the other .NET SDK tests where we haven't noticed the dupes for a while due to the similar implementation.

Reply:
This is a test helper method and it can be used multiple times during a single test (isn't yet, but there will be a next iteration of changes for supporting Request tracking).
I get your point though so I've added an assertion to check that only a single item is added to the ```sentItems``` list